### PR TITLE
Remove non-transparent requirement added to prerelease gems

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
## What was the end-user or developer problem that led to this PR?

I recall this requirement confusing me while working in the resolver, since it could lead to different resolution process when working in master (2.5.0.dev currently) than when working with a released version.

It feels non transparent and too magical. I think we can safely assume these days that all RubyGems and Bundler versions that will ever bundle a new gem created in 2023 support prereleases.

So this non transparent requirement is not necessary. It should be the gem author to explicitly add this kind of constraint, not RubyGems. Setting the version of a gem should do just that, set the version.

This has also caused Circular require issues in the past, and has special conditions attached to workaround that.

## What is your fix for the problem, implemented in this PR?

Remove the extra constraint.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Review by hsbt — APPROVED** [review]

File: lib/rubygems/specification.rb
Comment (martinemde) [inline lib/rubygems/specification.rb:2686]: Need to update the comment on `def version=` above as well.

**Review by martinemde — APPROVED** [review]
It seems extremely unlikely that this would still be necessary. That is an old RubyGems. Just update the comment before merging.

**Comment by martinemde** [comment]
Ready to merge after the comment is fixed. (just a note to self, since I almost came back and merged this)

File: lib/rubygems/specification.rb
Comment (deivid-rodriguez) [reply lib/rubygems/specification.rb:2686]: Thanks for catching that!

**Pull request merged**
